### ISSUE TITLE
Site Provisioning fails when trying to add event receiver 'LocationBasedMetadataDefaultsReceiver' to a list

### ISSUE THREAD

**Issue by nicolaor** [pnp/pnpframework#838]
During site provisioning on one particular tenant we have identified the following issue:

When Lists are being configured, the step FieldAndListProvisioningStepHelper.Step.ListSettings fails when setting the LocationBasedMetadataDefaultsReceiver event receiver on the list.
The issue appeared suddenly on a particular tenant and can be reproduced with any list or library only on that tenant.

The exception thrown is "Microsoft.SharePoint.Client.ServerUnauthorizedAccessException" as can be seen in the below screenshot.
 
![image](https://user-images.githubusercontent.com/89914175/219379051-63b22d7e-6661-4461-aee5-930c1eaf7ff2.png)

As a temporary workaround, we changed the code in PnP.Framework on our end to catch that specific exception and not rethrow it further.

Does somebody know if it is still necessary to add that particular event receiver on every list ? 
If a standard list is created via UI that event receiver is not registered. Instead we see the following event receivers (all webhook based and asynchronous)

![image](https://user-images.githubusercontent.com/89914175/219388222-dc38f328-f6c7-42e7-b25c-e66e84e344ac.png)

Could it be that Microsoft is deprecating the old synchronous event receivers and has rolled out a change on some tenants which is causing the issue ? @jansenbe do you have any information about this ?

**Comment by patrikhellgren** [comment]
@nicolaor I have been troubleshooting this exact same issue the last couple of days so good to at least see that someone else is also seeing this. At the moment we see it in just one customer tenant and it started there sometime over the weekend. This does not appear only when using provisioning and you can reproduce it using e.g. Set-PnPColumnDefaultValues or just adding the event receiver using PowerShell.

I would also believe that when you see no event receiver created when using the UI to set column default values, it will not work when uploading documents and no values will be set for the document.

In my testing I have found the following:
- If I create a new document library and enable a column default value using the UI everything works as expected and the event receiver is also added. After this when uploading a new document the field value is set correctly on the document.
- If I create a new document library and use provisioning or Set-PnPColumnDefaultValues logged in as a user with Site Collection Admin permissions to set a default value in the library it will fail with an Unauthorized exception when adding the event receiver. The field value will still be set in the client_LocationBasedDefaults.html file but new documents will not have the field value set. After this happens, if I use the UI to set a column default value in the same library it will not add the event receiver either. So it seems like after we have got an Unauthorized exception when trying to add the event receiver the library cannot have the event receiver ever created after that.
- Another thing I have seen, related or not, is that in the problematic tenant I cannot use CSOM with ACS App-only permissions to get any objects (web, lists etc). The exception here is also Unauthorized even though the app has full control permissions to all sites in the tenant. 

I know that there has been for many years a known issue with event receivers added by an app with app-only permissions and that the same event receiver would have to be removed by an app with the same client id or else it would throw an Unauthorized exception so that is why I connect the dots between the bullet points above.

@jansenbe Can you take this internally on your side  or should we post this in the sp-dev-docs repo also since it is a SharePoint developer issue and probably not related to PnP.Framework?

**Comment by jackpoz** [comment]
This issue has been reported also on PnP PowerShell repo at https://github.com/pnp/powershell/issues/2814

**Comment by jackpoz** [comment]
I submitted https://github.com/pnp/pnpframework/pull/840 that will allow to apply those templates that don't try to add location-based default values, which should be the majority of templates.
Those templates that have location-based default values will still fail.

**Comment by magarma** [comment]
Any update on this? We are facing the same issue in several production tenants and it seems that something is spreading out.
@jansenbe

**Comment by fzbm** [comment]
This already two weeks old issue finally hit one of our tenants too. This lets me assume, that there is an ongoing rollout of an update in progress and more and more tenants will be affected by this issue soon.

I'm currently testing if our application works fine with the latest nightly build of PnP Framework, but the PR #840 is only a band aid, because the error will reoccur as soon as PnP has to set a default value (as already mentioned by @jackpoz in that PR).

It would be nice if one of the (Microsoft) PnP team members could reach out to the SharePoint guys (using the unbureaucratic channel) and clarify the reason for this issue.

**Comment by magarma** [comment]
Just curiosity.... checking this version.... in the tenant that fails gives this number...

https://XXXXXXXX.sharepoint.com/_vti_pvt/service.cnf

vti_encoding:SR|utf8-nl
vti_extenderversion:SR|16.0.0.12003

However, others tenant where is not failing have 
vti_extenderversion:SR|16.0.0.23415


For those that are having the issue....what version do you have?

**Comment by fzbm** [comment]
Have checked that and for the tenant which failed until I deployed the hotfix the version is `16.0.0.23415`. I also found one tenant with version `16.0.0.23408` but I cannot say, if the issue would have occurred there too.

Is also found a tenant which is using the `12003` version, which I find a little bit strange. I mean there is a huge gap between `12000` and `23000`. Is there a reason for this big difference?
